wled00/xml.cpp

```diff
@@ -264,50 +312,72 @@ void getSettingsJS(byte subPage, char* dest)
     }
   }
 
-  if (subPage == 2) {
+  if (subPage == 2)
+  {
     char nS[8];
 
     // add reserved and usermod pins as d.um_p array
+    oappend(SET_F("d.um_p=[6,7,8,9,10,11"));
+
     DynamicJsonDocument doc(JSON_BUFFER_SIZE/2);
     JsonObject mods = doc.createNestedObject(F("um"));
     usermods.addToConfig(mods);
-    oappend(SET_F("d.um_p=["));
-    if (!mods.isNull()) {
-      uint8_t i=0;
-      for (JsonPair kv : mods) {
-        if (!kv.value().isNull()) {
-          // element is an JsonObject
-          JsonObject obj = kv.value();
-          if (obj["pin"] != nullptr) {
-            if (obj["pin"].is<JsonArray>()) {
-              JsonArray pins = obj["pin"].as<JsonArray>();
-              for (JsonVariant pv : pins) {
-                if (i++) oappend(SET_F(","));
-                oappendi(pv.as<int>());
-              }
-            } else {
-              if (i++) oappend(SET_F(","));
-              oappendi(obj["pin"].as<int>());
-            }
+    if (!mods.isNull()) fillUMPins(mods);
+
+    #ifdef WLED_ENABLE_DMX
+      oappend(SET_F(",2")); // DMX hardcoded pin
+    #endif
+
+    #ifdef WLED_ENABLE_ADALIGHT
```

Comment:
Can be removed. Prohibiting pin 3 just because Adalight could be received is not reasonable.
Adalight receiving code will not allocate the pin. Serial data in is not something we can control/manage in WLED code (with the exception of not sending `Ada` on boot if pin 3 is allocated), it is sent by the computer "just like that".

Reply:
Just checked, Adalight/ Serial JSON receiving code is now disabled when the pin is allocated :) For most users, pin 3 availability is more crucial than Adalight functionality.

Reply:
I opted to have Ada light compile in optionally instead of default.
Might consider the same for master (only change wled.h)

Reply:
Let's have it compiled in by default. The main use case is people using the instance as an ambilight with just a single bus on pin 2/16.
Instead, what do you think about a warning like this for LED/UM settings?
`⚠ Using pin 3. Adalight and Serial JSON will be disabled.`

Reply:
This kind of approach adds whole lot of issues IMO.
One workaround could be to add checkbox somewhere in settings for enabling disabling Ada/Serial.

Reply:
No need for a checkbox, it is automatically disabled once pin 3 is allocated for anything else:
https://github.com/Aircoookie/WLED/blob/master/wled00/wled_serial.cpp#L24

A note in kno.wled.ge might also be sufficient to let users know that configuring pin 3 will disable serial input.

Reply:
Ok. 😄